joomla/joomla-cms
# User enumeration in the "lost password" and "lost username" views

### Summary of Changes

> The "lost password" and "lost username" allow attackers to verify the existence of a given username because they respond with different error messages depending if a username exists or not.

### Testing Instructions

- apply this patch
- try to remind the user name (with an invalid email)
- try to reset the PW. (with an invalid email)
- verify the generic message.
- enable debug
- use a not existing mail
- verify the original error message before the patch.

### Expected result

After the patch we give a generic message `If the email address was correct a reminder has been sent. Please check your mail.` || `If the email address was correct a new password has been sent. Please check your mail`. The difference / error message is only displayed if you have enabled Debug mode in the Global Config.

### Actual result

Before the patch if you enter a existing email you are redirected if not you stay on the same page. This means everyone can verify the existence of a given username / email

### Documentation Changes Required

None.

### Additional infos

This is a issue that was reported as security issue to the JSST. But after discussion the JSST agrees this should be discussed in the public tracker as it may can harm the usibility.

cc: @joomla/security
